File: lib/api/session.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

const API_BASE = "https://mangalam-wifi-zone.onrender.com";

Future<void> startSession(String mac, Map plan) async {
  final res = await http.post(
    Uri.parse("$API_BASE/api/v1/session/start"),
    headers: {"Content-Type": "application/json"},
    body: jsonEncode({"mac": mac, "plan": plan}),
  );

  if (res.statusCode != 200) {
    throw Exception('Failed to start session: ${res.body}');
  }
}

File: web_client/src/api/session.js
const API_BASE = import.meta.env.VITE_API_BASE_URL || "";

export async function startSession(mac, plan) {
  const res = await fetch(`${API_BASE}/api/v1/session/start`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ mac, plan }),
  });

  if (!res.ok) {
    const errorData = await res.json().catch(() => ({ message: "An unknown error occurred" }));
    throw new Error(errorData.message || "Failed to start session");
  }

  return res.json();
}
